core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -0,0 +1,5534 @@
+
+      // is date parsing lenient?
+      //
+      dateFormatLenient = inputStream.readBoolean();
+
+      // What is the date format locale?
+      //
+      String strDateFormatLocale = readString( inputStream );
+      if ( Utils.isEmpty( strDateFormatLocale ) ) {
+        dateFormatLocale = null;
+      } else {
+        dateFormatLocale = EnvUtil.createLocale( strDateFormatLocale );
+      }
+
+      // What is the time zone to use for date parsing?
+      //
+      String strTimeZone = readString( inputStream );
+      if ( Utils.isEmpty( strTimeZone ) ) {
+        dateFormatTimeZone = TimeZone.getDefault();
+      } else {
+        dateFormatTimeZone = EnvUtil.createTimeZone( strTimeZone );
+      }
+
+      // is string to number parsing lenient?
+      lenientStringToNumber = inputStream.readBoolean();
+    } catch ( EOFException e ) {
+      throw new KettleEOFException( e );
+    } catch ( IOException e ) {
+      throw new KettleFileException( toString() + " : Unable to read value metadata from input stream", e );
+    }
+
+  }
+
+  @Override
+  public String getMetaXML() throws IOException {
+    StringBuilder xml = new StringBuilder();
+
+    xml.append( XMLHandler.openTag( XML_META_TAG ) );
+
+    xml.append( XMLHandler.addTagValue( "type", getTypeDesc() ) );
+    xml.append( XMLHandler.addTagValue( "storagetype", getStorageTypeCode( getStorageType() ) ) );
+
+    switch ( storageType ) {
+      case STORAGE_TYPE_INDEXED:
+        xml.append( XMLHandler.openTag( "index" ) );
+
+        // Save the indexed strings...
+        //
+        if ( index != null ) {
+          for ( int i = 0; i < index.length; i++ ) {
+            try {
+              switch ( type ) {
+                case TYPE_STRING:
+                  xml.append( XMLHandler.addTagValue( "value", (String) index[i] ) );
+                  break;
+                case TYPE_NUMBER:
+                  xml.append( XMLHandler.addTagValue( "value", (Double) index[i] ) );
+                  break;
+                case TYPE_INTEGER:
+                  xml.append( XMLHandler.addTagValue( "value", (Long) index[i] ) );
+                  break;
+                case TYPE_DATE:
+                  xml.append( XMLHandler.addTagValue( "value", (Date) index[i] ) );
+                  break;
+                case TYPE_BIGNUMBER:
+                  xml.append( XMLHandler.addTagValue( "value", (BigDecimal) index[i] ) );
+                  break;
+                case TYPE_BOOLEAN:
+                  xml.append( XMLHandler.addTagValue( "value", (Boolean) index[i] ) );
+                  break;
+                case TYPE_BINARY:
+                  xml.append( XMLHandler.addTagValue( "value", (byte[]) index[i] ) );
+                  break;
+                default:
+                  throw new IOException( toString() + " : Unable to serialize index storage type to XML for data type "
+                      + getType() );
+              }
+            } catch ( ClassCastException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)